scripts/generate-patch

```diff
@@ -51,7 +51,7 @@ for f in packages/*; do
 					helm_repo=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].repository" | head -n 1)
 					# Remove trailing slash
 					[[ ${helm_repo} == */ ]] && helm_repo=${helm_repo%/}
-					if [[ ${helm_repo} == file://* ]] || [[ ${version} == *'*'* ]]; then
+					if [[ ${helm_repo} == file://* ]] || [[ ! ${version} =~ ^[0-9\.]+$ ]]; then
```

Comment:
Yes, the previous version of the code was just checking to see if a wildcard character in the version. This version of the code checks to see that the version provided only consists of digits (`0-9`) or periods (`\.`), since it is possible / valid to specify a version string that has a non-fixed version.

It's not the best solution, since there are cases like [datadog](https://github.com/helm/charts/blob/ef5b8687622cce8d6b0c7ee1df80329603738c2a/stable/datadog/requirements.yaml#L3) where you are specifying a fixed version number but this script won't pick it up as expected, but instead of introducing a more complicated expression I kept it simple to unblock the failing CI. Properly validating the semver and checking can be done in https://github.com/rancher/rancher/issues/28730.